sympy/core/add.py

```diff
@@ -835,10 +833,29 @@ def _eval_as_leading_term(self, x):
             return expr.as_leading_term(x)
 
         infinite = [t for t in expr.args if t.is_infinite]
+        compute = False
+
+        leading_term = [t.as_leading_term(x) for t in expr.args]
+        new_expr = expr.func(*leading_term).removeO()
+
+        if not new_expr:
+            compute = True
+        elif new_expr is not S.NaN:
```

Comment:
What should be done if new_expr is nan?

Reply:
then ```compute``` will be ```False```. And ```old.func._from_args(infinite)``` will be executed(line 861-861)